testutils/workload: TestSetup/rows=10/batch=100 failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/0d0d9f1be15ceb3a02e9018106042fa911ede352

Parameters:
```
TAGS=
GOFLAGS=
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=486317&tab=buildLog

```
    	workload_test.go:68: pq: relation "bank" does not exist


ERROR: exit status 1

context canceled
make: *** [stress] Error 1
Makefile:802: recipe for target 'stress' failed
```

--- a/pkg/sql/drop_test.go
+++ b/pkg/sql/drop_test.go
@@ -827,6 +827,23 @@ func TestDropAndCreateTable(t *testing.T) {
 	}
 }
 
+func TestDropAndCreateDatabase(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	t.Skip(`#22256`)
+
+	ctx := context.Background()
+	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
+	defer s.Stopper().Stop(ctx)
+	sqlDB := sqlutils.MakeSQLRunner(db)
+
+	for i := 0; i < 20; i++ {
+		sqlDB.Exec(t, `DROP DATABASE IF EXISTS test`)
+		sqlDB.Exec(t, `CREATE DATABASE test`)
+		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY)`)
+		sqlDB.Exec(t, `INSERT INTO foo VALUES (1), (2), (3)`)
+	}
+}
+
 // Test commands while a table is being dropped.
 func TestCommandsWhileTableBeingDropped(t *testing.T) {
 	defer leaktest.AfterTest(t)()

--- a/pkg/testutils/workload/bank/bank_test.go
+++ b/pkg/testutils/workload/bank/bank_test.go
@@ -25,7 +25,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 )
 
-func TestSplit(t *testing.T) {
+func TestBank(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
 	tests := []struct {
@@ -41,18 +41,17 @@ func TestSplit(t *testing.T) {
 	}
 
 	ctx := context.Background()
-	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
 	defer s.Stopper().Stop(ctx)
+	sqlutils.MakeSQLRunner(db).Exec(t, `CREATE DATABASE test`)
 
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("rows=%d/ranges=%d", test.rows, test.ranges), func(t *testing.T) {
 			sqlDB := sqlutils.MakeSQLRunner(db)
+			sqlDB.Exec(t, `DROP TABLE IF EXISTS bank`)
 
 			bank := FromConfig(test.rows, defaultPayloadBytes, test.ranges)
 			bankTable := bank.Tables()[0]
-			sqlDB.Exec(t, `DROP DATABASE IF EXISTS data CASCADE`)
-			sqlDB.Exec(t, `CREATE DATABASE data`)
-			sqlDB.Exec(t, `USE data`)
 			sqlDB.Exec(t, fmt.Sprintf(`CREATE TABLE %s %s`, bankTable.Name, bankTable.Schema))
 
 			if err := Split(sqlDB.DB, bank); err != nil {

--- a/pkg/testutils/workload/workload_test.go
+++ b/pkg/testutils/workload/workload_test.go
@@ -54,12 +54,12 @@ func TestSetup(t *testing.T) {
 	ctx := context.Background()
 	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
 	defer s.Stopper().Stop(ctx)
+	sqlutils.MakeSQLRunner(db).Exec(t, `CREATE DATABASE test`)
 
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("rows=%d/batch=%d", test.rows, test.batchSize), func(t *testing.T) {
 			sqlDB := sqlutils.MakeSQLRunner(db)
-			sqlDB.Exec(t, `DROP DATABASE IF EXISTS test`)
-			sqlDB.Exec(t, `CREATE DATABASE test`)
+			sqlDB.Exec(t, `DROP TABLE IF EXISTS bank`)
 
 			gen := bank.FromRows(test.rows)
 			if _, err := workload.Setup(sqlDB.DB, gen, test.batchSize); err != nil {
@@ -86,16 +86,16 @@ func TestSplits(t *testing.T) {
 	ctx := context.Background()
 	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{UseDatabase: `test`})
 	defer s.Stopper().Stop(ctx)
+	sqlutils.MakeSQLRunner(db).Exec(t, `CREATE DATABASE test`)
 
 	for _, ranges := range tests {
 		t.Run(fmt.Sprintf("ranges=%d", ranges), func(t *testing.T) {
 			sqlDB := sqlutils.MakeSQLRunner(db)
-			sqlDB.Exec(t, `DROP DATABASE IF EXISTS test`)
-			sqlDB.Exec(t, `CREATE DATABASE test`)
+			sqlDB.Exec(t, `DROP TABLE IF EXISTS bank`)
 
 			gen := bank.FromConfig(rows, payloadBytes, ranges)
 			table := gen.Tables()[0]
-			sqlDB.Exec(t, fmt.Sprintf(`CREATE TABLE test.%s %s`, table.Name, table.Schema))
+			sqlDB.Exec(t, fmt.Sprintf(`CREATE TABLE %s %s`, table.Name, table.Schema))
 
 			if err := workload.Split(ctx, sqlDB.DB, table, concurrency); err != nil {
 				t.Fatalf("%+v", err)

